packages/core/src/types/UnaryOperation.ts

```diff
@@ -1,3 +1,4 @@
-export type UnaryOperation<TInput, TOutput = TInput> = (
-  value: TInput
+export type UnaryOperation<TInput, TParams = readonly [], TOutput = TInput> = (
+  operand: TInput,
+  ...rest: TParams extends readonly [...infer TRest] ? TRest : never
```

Comment:
@sarahdayan what do you think of this? I would also change BinaryOperation to be simular. Without something like this adding return types to all the core functions is really messy looking and receptive.